src/plugins/data/public/autocomplete/providers/value_suggestion_provider.ts

```diff
@@ -69,6 +86,9 @@ export const setupValueSuggestionProvider = (core: CoreSetup): ValueSuggestionsG
       return [];
     }
 
-    return await requestSuggestions(title, field, query, boolFilter, signal);
+    const timeFilter = useTimeRange ? getAutocompleteTimefilter(indexPattern) : undefined;
+    const filterQuery = timeFilter ? buildQueryFromFilters([timeFilter], indexPattern).filter : [];
+    const filters = [...(boolFilter ? boolFilter : []), ...filterQuery];
```

Comment:
APM, Uptime and others are already applying the date range + other filters via the `boolFilter`. Perhaps the component shouldn't re-apply he date filter if `boolFilter` is defined?

Reply:
I see only one place using the filter explicitly (other than `uptime` which I had already fixed), and that's in `x-pack/plugins/apm/public/components/shared/KueryBar/index.tsx` and I think the solution is actually related to https://github.com/elastic/observability-dev/issues/1139.

If a solution is using the global `timefilter` service, then it should rely on `autocomplete` service fetching it internally.
If a solution is not using the global `timefilter` service, it should pass `useTimeFilter: false` and pass in any filters it wants using `boolFilters`.

Does the code in that file already rely on that service?

Reply:
> Does the code in that file already rely on that service?

APM uses the timefilter API's if that's what you mean? Eg. `plugins.data.query.timefilter.timefilter.setTime`. But we also have additional filters. So it sounds like we should remove the time range from the filters we pass to `getQuerySuggestions` (via `boolFilter` prop).